components/Signatures.js

```diff
@@ -15,15 +15,19 @@ const cleanHandle = (handle, address, verified) => {
   return address.slice(0, 8);
 }
 
-export default function Signatures({txId, sigs, setSigs}) {
+export default function Signatures({txId, sigs, setSigs,signed}) {
   const [cursor, setCursor] = React.useState("")
   const [sortedSigs, setSortedSigs] = React.useState([])
   const [reachedEnd, setReachedEnd] = React.useState(false)
 
   React.useEffect(() => {
     setCursor(sigs[sigs.length-1] && sigs[sigs.length-1].CURSOR)
     setSortedSigs(sortSigs(dedupe(sigs)))
-  }, [sigs])
+  }, [sigs]);
+
+  React.useEffect(() => {
```

Comment:
If we assume the signature will take a few minutes to be in Arweave (because the transaction has to confirm), I don't think this will work. We may need to inject a mocked version of the signature into the list rather than fetching. What do you think?

Reply:
What do you think of my new code?